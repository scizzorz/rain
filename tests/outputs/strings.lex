keyword 'var'
name 'main'
symbol '='
keyword 'func'
symbol '('
symbol ')'
indent
keyword 'var'
name 's'
symbol '='
string 'Hello'
newline
keyword 'var'
name 'p'
symbol '='
string 'world'
newline
keyword 'var'
name 'sp'
symbol '='
name 's'
operator '$'
string ', '
operator '$'
name 'p'
newline
name 'print'
symbol '('
name 'sp'
symbol ')'
newline
keyword 'var'
name 'i'
symbol '='
operator '-'
name 'sp'
symbol ':'
name 'length'
symbol '('
symbol ')'
operator '-'
int 1
newline
keyword 'while'
name 'i'
operator '<'
name 'sp'
symbol ':'
name 'length'
symbol '('
symbol ')'
operator '+'
int 1
indent
name 'print'
symbol '('
string '['
operator '$'
name 'tostr'
symbol '('
name 'i'
symbol ')'
operator '$'
string '] = '
operator '$'
name 'tostr'
symbol '('
name 'sp'
symbol '['
name 'i'
symbol ']'
symbol ')'
symbol ')'
newline
name 'i'
symbol '='
name 'i'
operator '+'
int 1
newline
dedent
newline
name 'print'
symbol '('
name 'sp'
symbol '['
symbol '['
int 0
symbol ','
int 4
symbol ']'
symbol ']'
symbol ')'
newline
name 'print'
symbol '('
name 'sp'
symbol '['
symbol '['
int 2
symbol ','
int 6
symbol ']'
symbol ']'
symbol ')'
newline
name 'print'
symbol '('
name 'sp'
symbol '['
symbol '['
int 2
symbol ','
int -1
symbol ']'
symbol ']'
symbol ')'
newline
name 'print'
symbol '('
name 'sp'
symbol '['
symbol '['
int -4
symbol ','
int -1
symbol ']'
symbol ']'
symbol ')'
newline
name 'print'
symbol '('
name 'sp'
symbol '['
symbol '['
int -8
symbol ','
int 8
symbol ']'
symbol ']'
symbol ')'
newline
name 'print'
symbol '('
name 'sp'
symbol '['
symbol '['
int 2
symbol ','
name 'sp'
symbol ':'
name 'length'
symbol '('
symbol ')'
symbol ']'
symbol ']'
symbol ')'
newline
name 'print'
symbol '('
name 'sp'
symbol '['
symbol '['
int 2
symbol ','
name 'sp'
symbol ':'
name 'length'
symbol '('
symbol ')'
operator '+'
int 2
symbol ']'
symbol ']'
symbol ')'
newline
name 'print'
symbol '('
name 'sp'
symbol '['
symbol '['
int 15
symbol ','
int 12
symbol ']'
symbol ']'
symbol ')'
newline
name 'print'
symbol '('
name 'sp'
symbol '['
symbol '['
int 4
symbol ','
int 3
symbol ']'
symbol ']'
symbol ')'
newline
dedent
newline
EOF
