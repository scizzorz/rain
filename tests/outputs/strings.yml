!program;1
stmts:
- !assn;3
  export: false
  lhs: !name;1
    value: main
  rhs: !func;2
    body: !block;2
      expr: null
      stmts:
      - !assn;3
        export: false
        lhs: !name;1
          value: s
        rhs: !str;1
          value: Hello
        var: true
      - !assn;3
        export: false
        lhs: !name;1
          value: p
        rhs: !str;1
          value: world
        var: true
      - !assn;3
        export: false
        lhs: !name;1
          value: sp
        rhs: !binary;1
          lhs: !binary;1
            lhs: !name;1
              value: s
            op: $
            rhs: !str;1
              value: ', '
          op: $
          rhs: !name;1
            value: p
        var: true
      - !call;1
        args:
        - !name;1
          value: sp
        catch: false
        func: !name;1
          value: print
      - !assn;3
        export: false
        lhs: !name;1
          value: i
        rhs: !binary;1
          lhs: !unary;1
            op: '-'
            val: !method;1
              args: []
              catch: false
              lhs: !name;1
                value: sp
              rhs: !str;1
                value: length
          op: '-'
          rhs: !int;1
            value: 1
        var: true
      - !while;1
        body: !block;2
          expr: null
          stmts:
          - !call;1
            args:
            - !binary;1
              lhs: !binary;1
                lhs: !binary;1
                  lhs: !str;1
                    value: '['
                  op: $
                  rhs: !call;1
                    args:
                    - !name;1
                      value: i
                    catch: false
                    func: !name;1
                      value: tostr
                op: $
                rhs: !str;1
                  value: '] = '
              op: $
              rhs: !call;1
                args:
                - !index;1
                  lhs: !name;1
                    value: sp
                  rhs: !name;1
                    value: i
                catch: false
                func: !name;1
                  value: tostr
            catch: false
            func: !name;1
              value: print
          - !assn;3
            export: false
            lhs: !name;1
              value: i
            rhs: !binary;1
              lhs: !name;1
                value: i
              op: +
              rhs: !int;1
                value: 1
            var: false
        pred: !binary;1
          lhs: !name;1
            value: i
          op: <
          rhs: !binary;1
            lhs: !method;1
              args: []
              catch: false
              lhs: !name;1
                value: sp
              rhs: !str;1
                value: length
            op: +
            rhs: !int;1
              value: 1
      - !call;1
        args:
        - !index;1
          lhs: !name;1
            value: sp
          rhs: !array;1
            items:
            - !int;1
              value: 0
            - !int;1
              value: 4
        catch: false
        func: !name;1
          value: print
      - !call;1
        args:
        - !index;1
          lhs: !name;1
            value: sp
          rhs: !array;1
            items:
            - !int;1
              value: 2
            - !int;1
              value: 6
        catch: false
        func: !name;1
          value: print
      - !call;1
        args:
        - !index;1
          lhs: !name;1
            value: sp
          rhs: !array;1
            items:
            - !int;1
              value: 2
            - !int;1
              value: -1
        catch: false
        func: !name;1
          value: print
      - !call;1
        args:
        - !index;1
          lhs: !name;1
            value: sp
          rhs: !array;1
            items:
            - !int;1
              value: -4
            - !int;1
              value: -1
        catch: false
        func: !name;1
          value: print
      - !call;1
        args:
        - !index;1
          lhs: !name;1
            value: sp
          rhs: !array;1
            items:
            - !int;1
              value: -8
            - !int;1
              value: 8
        catch: false
        func: !name;1
          value: print
      - !call;1
        args:
        - !index;1
          lhs: !name;1
            value: sp
          rhs: !array;1
            items:
            - !int;1
              value: 2
            - !method;1
              args: []
              catch: false
              lhs: !name;1
                value: sp
              rhs: !str;1
                value: length
        catch: false
        func: !name;1
          value: print
      - !call;1
        args:
        - !index;1
          lhs: !name;1
            value: sp
          rhs: !array;1
            items:
            - !int;1
              value: 2
            - !binary;1
              lhs: !method;1
                args: []
                catch: false
                lhs: !name;1
                  value: sp
                rhs: !str;1
                  value: length
              op: +
              rhs: !int;1
                value: 2
        catch: false
        func: !name;1
          value: print
      - !call;1
        args:
        - !index;1
          lhs: !name;1
            value: sp
          rhs: !array;1
            items:
            - !int;1
              value: 15
            - !int;1
              value: 12
        catch: false
        func: !name;1
          value: print
      - !call;1
        args:
        - !index;1
          lhs: !name;1
            value: sp
          rhs: !array;1
            items:
            - !int;1
              value: 4
            - !int;1
              value: 3
        catch: false
        func: !name;1
          value: print
    params: []
    rename: null
  var: true
