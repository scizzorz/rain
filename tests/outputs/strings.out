Hello, world
[-13] = null
[-12] = H
[-11] = e
[-10] = l
[-9] = l
[-8] = o
[-7] = ,
[-6] =  
[-5] = w
[-4] = o
[-3] = r
[-2] = l
[-1] = d
[0] = H
[1] = e
[2] = l
[3] = l
[4] = o
[5] = ,
[6] =  
[7] = w
[8] = o
[9] = r
[10] = l
[11] = d
[12] = null
Hell
llo,
llo, worl
orl
o, w
llo, world
llo, world


